src/connectedvmware/azext_connectedvmware/custom.py

```diff
@@ -987,6 +994,103 @@ def create_vm(
+| extend p=parse_json(properties)
+| extend p=parse_json(properties)
+| extend moName = tolower(tostring(p['moName']))
+| extend moRefId = tolower(tostring(p['moRefId']))
+| where moName =~ '{mo_name}'
+| project id, moName, moRefId, props=p
+"""
+
+        query = " ".join(query.splitlines())
+        query_request = QueryRequest(
+            subscriptions=[get_subscription_id(cmd.cli_ctx)],
+            query=query,
+        )
+        query_response: QueryResponse = arg_client.resources(query_request)
+        vm_list = []
+        for vm in query_response.data:
+            vm_list.append(vm)
+        if len(vm_list) == 0:
+            raise ResourceNotFoundError(
+                f"Inventory item with mo_name '{mo_name}' not found in vCenter '{vcenter}'."
+            )
+        if len(vm_list) > 1:
+            # Ask user to select one of the inventory items.
+            # print the json of the inventory items and ask user to select one.
+            logger = get_logger(__name__)
+            for i, vm in enumerate(vm_list):
+                logger.warning(
+                    "[%s] %s \n%s",
+                    i + 1, vm["id"], json.dumps(vm["props"], indent=2)
+                )
+            logger.warning(
+                "Multiple inventory items found with mo_name '%s' in vCenter '%s'. "
+                "Please check the logs above for the properties of the inventory items.",
+                mo_name, vcenter
+            )
+            selected = prompt_y_n(
+                "Do you want to select one of the inventory items to proceed?"
+            )
+            if not selected:
+                raise InvalidArgumentValueError(
+                    "Please provide a unique mo_name or inventory_item id. "
+                    "Multiple inventory items found with the same mo_name, and no selection was made."
+                )
+            # print just the inventory item id in a list and ask user to select one.
+            for i, vm in enumerate(vm_list):
+                logger.warning(
+                    "[%s] %s",
+                    i + 1, vm["id"]
+                )
+            selected = prompt(
+                f"Select one of the inventory items by entering the index between 1 and {len(vm_list)}: "
+            )
+            try:
+                while True:
+                    selected = int(selected.strip())
+                    if 1 <= selected <= len(vm_list):
+                        logger.warning(
+                            "You selected: %s",
+                            vm_list[selected - 1]["id"]
+                        )
+                        break
+                    selected = prompt(
+                        f"Invalid index. Please enter a valid index between 1 and {len(vm_list)}: "
+                    )
+            except ValueError:
+                raise InvalidArgumentValueError(
+                    "Invalid input. Please provide a valid index."
```

Comment:
Let us validate this code path.

Reply:
Have done the validation, and added the logs in the PR description.